{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Input:\n",
    "python aligned_footprinting_data.sgr motifxofxinterest +/-bp  \n",
    "python sys.argv[1]                   sys.argv[2]       sys.argv[3]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Import requisite libraries\n",
    "import math\n",
    "import sys\n",
    "import os\n",
    "import time\n",
    "import numpy as np\n",
    "from decimal import *\n",
    "import pandas as pd\n",
    "import re"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# read in file\n",
    "# -All *Norm.sgr files have structure:\n",
    "# -[Chromosome] [bp #] [normalized count]\n",
    "#   \tchrI\t\t0\t\t\t0\n",
    "#   \tchrI\t\t1\t\t\t0\n",
    "#   \tchrI\t\t2\t\t\t0\n",
    "#   \tchrI\t\t3\t\t\t0\n",
    "#   \tchrI\t\t4\t\t\t0\n",
    "\n",
    "#mnasedata = sys.argv[1] \n",
    "#motif = sys.argv[2] \n",
    "#bp = sys.argv[3]\n",
    "\n",
    "mnasedata = \"YS001Q_MNase_Norm_SHORT.sgr\" \n",
    "motif = 'CACC'\n",
    "bp = 500\n",
    "              \n",
    "\n",
    "# Norml is a list of lists of two values each. First value is bp#, second value is normalized count #\n",
    "Norml = []\n",
    "\n",
    "with open (\"YS001Q_MNase_Norm_SHORT.sgr\") as f:\n",
    "    for line in f:\n",
    "        values = line.split()\n",
    "        Norml.append(([(values[1]),(values[2])]))\n",
    "        \n",
    "Norma = np.array(Norml)\n",
    "\n",
    "# to make faster: key lines to chromosomes in dictionary\n",
    "# store integers - smaller than strings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# find motifxofxinterest IN REFERENCE GENOME (Watson strand)\n",
    "\n",
    "## Make sequence array\n",
    "# yeast_sequence.sgr\n",
    "## int[chromosome] int[bp#] chr[base]\n",
    "\n",
    "refl = []\n",
    "\n",
    "with open (\"yeast_sequence_SHORT.sgr\") as f:\n",
    "    for line in f:\n",
    "        values = line.split()\n",
    "        refl.append([int(values[0]),int(values[1]),values[2]])\n",
    "\n",
    "refa = np.array(refl)\n",
    "\n",
    "# note: everything in this array is now a string.\n",
    "\n",
    "### Note to self: chr() turns an INTEGER into a CHARACTER. If it's already a character, don't do anything.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Import motif as array\n",
    "\n",
    "## somehow get sys.argv[2] into python AND read the length\n",
    "## for now the input is this motif string\n",
    "motif = 'CACC'\n",
    "# motif = 'GGTG' \n",
    "# This works too!!\n",
    "\n",
    "motifl = []\n",
    "\n",
    "for i in range(len(motif)):\n",
    "    motifl.append(motif[i])\n",
    "motifa = np.array(motifl)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['G' 'G' 'T' 'G']\n"
     ]
    }
   ],
   "source": [
    "# find motifxofxinterest in Crick strand\n",
    "\n",
    "bases = {'A':'T','T':'A','C':'G','G':'C'}\n",
    "\n",
    "def reverse_complement(r):\n",
    "    output = []\n",
    "    for i in range(len(r)):\n",
    "        output.append(bases[r[i]])\n",
    "#    return ((\"\".join(output))[::-1])\n",
    "    return ((output)[::-1])\n",
    "motifrc = reverse_complement(motif)\n",
    "motifrca = np.array(motifrc)\n",
    "print(motifrca)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Roman numeral converter\n",
    "#https://www.w3resource.com/python-exercises/class-exercises/python-class-exercise-1.php\n",
    "def int_to_roman(num):\n",
    "        val = [\n",
    "            1000, 900, 500, 400,\n",
    "            100, 90, 50, 40,\n",
    "            10, 9, 5, 4,\n",
    "            1\n",
    "            ]\n",
    "        syb = [\n",
    "            \"M\", \"CM\", \"D\", \"CD\",\n",
    "            \"C\", \"XC\", \"L\", \"XL\",\n",
    "            \"X\", \"IX\", \"V\", \"IV\",\n",
    "            \"I\"\n",
    "            ]\n",
    "        roman_num = ''\n",
    "        i = 0\n",
    "        while  num > 0:\n",
    "            for _ in range(num // val[i]):\n",
    "                roman_num += syb[i]\n",
    "                num -= val[i]\n",
    "            i += 1\n",
    "        return roman_num"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[['I', '+', '+', '+', '+', 3]]\n"
     ]
    }
   ],
   "source": [
    "### More original block\n",
    "#https://stackoverflow.com/questions/20763012/creating-a-pandas-dataframe-from-a-numpy-array-how-do-i-specify-the-index-colum\n",
    "df_ref = pd.DataFrame(refa, index=refa[:,0])\n",
    "\n",
    "df_motif = pd.DataFrame(motifa, index=refa[0:len(motifa),0])\n",
    "\n",
    "df_motifrc = pd.DataFrame(motifrca, index=refa[0:len(motifrca),0])\n",
    "\n",
    "location = []\n",
    "\n",
    "for i in range(len(refa)-len(motifa)):\n",
    "\n",
    "    if df_ref[i:i+len(motifa)][2].equals(df_motif[0:len(motifa)][0]):\n",
    "        chromosome_num = \"chr\" + int_to_roman(int(df_ref.iloc[i][0]))\n",
    "        location.append([chromosome_num[i], '+', '+', '+', '+', i])\n",
    "\n",
    "    elif df_ref[i:i+len(motifrca)][2].equals(df_motifrc[0:len(motifrca)][0]):\n",
    "        chromosome_num = \"chr\" + str(int_to_roman(int(df_ref.iloc[i][0])))\n",
    "        location.append([\"chr\"+chromosome_num, '-', '-', '-', '-', i])\n",
    "        \n",
    "print(location)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Sequence not found\n",
    "## look for possible mismatches"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "### USER INPUT ### \n",
    "#     Back into unix\n",
    "### Continue? ###\n",
    "#     take user input into code with sys.argv[]\n",
    "### Choose number of mismatches to tolerate ###"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## create list of iterations of motifxofxinterest with  bp off"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Choose mismatch to find ###"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# feed locations into something"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "part I took  0.0 seconds\n"
     ]
    },
    {
     "ename": "NameError",
     "evalue": "name 'A' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-34-68eff52c2658>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m     45\u001b[0m \u001b[0melapsed\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mtime\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mtime\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m-\u001b[0m\u001b[0mstart\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     46\u001b[0m \u001b[0mprint\u001b[0m \u001b[1;33m(\u001b[0m\u001b[1;34m'part I took '\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0melapsed\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m'seconds'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 47\u001b[1;33m \u001b[0mA\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mclose\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     48\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     49\u001b[0m \u001b[0mfile2\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mopen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mmotif\u001b[0m\u001b[1;33m+\u001b[0m\u001b[1;34m'_'\u001b[0m\u001b[1;33m+\u001b[0m\u001b[0mmnasedata\u001b[0m\u001b[1;33m+\u001b[0m\u001b[1;34m'_aligned'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mreadlines\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mNameError\u001b[0m: name 'A' is not defined"
     ]
    }
   ],
   "source": [
    "\n",
    "#List=open('/projects/mcknightlab/shared/Scripts_From_Hutch/Breeden/'+sys.argv[1]+'/'+sys.argv[1]+'_HITS_SITES.txt').readlines()\t\t#Select region of interest from features list and use prefix here\n",
    "#file=open(sys.argv[2]).readlines()\t\t#Input file 2 is a genomic coverage file of nucleosome positions\n",
    "#A=open(sys.argv[3]+'_'+sys.argv[1]+'_aligned', 'w')\t\t\t\t#Output file is the location of aligned positions\n",
    "\n",
    "# https://stackoverflow.com/questions/30333186/grab-part-of-filename-with-python\n",
    "# re = regular expressions\n",
    "# re.findall(\"ID_(\\d+).pdf\", fname)\n",
    "# \\d is data type. ASCII.\n",
    "\n",
    "A = []\n",
    "\n",
    "# To write A into a .sgv file later:\n",
    "#https://stackoverflow.com/questions/899103/writing-a-list-to-a-file-with-python\n",
    "#with open(re.findall(\"(\\d+)_Norm_*.sgr\", fname)+motif+'_', 'w') as f:\n",
    "#    for item in A:\n",
    "#        f.write(\"%s\\n\" % item)\n",
    " \n",
    "start=time.time()0\n",
    "\"\"\"\n",
    "Chr = list of locations of motif = location\n",
    "file = e = Norma\n",
    "f = chromosome\n",
    "g = position\n",
    "\"\"\"\n",
    "for j in range(len(Norma)):\t#Searches through entire normalized MNase coverage file\n",
    "\te=Norma[j] # e = each line in the MNase coverage file\n",
    "\tchromosome = e[0] # position 0 in each line is the chromosome\n",
    "\tposition = e[1] # position 1 in each line is the bp position\n",
    "\tif position in location: # if the bp position is present in the list of locations of the motif\n",
    "\t\tfor i in range(len(location)): # for each location of this motif in the genome\n",
    "\t\t\tb=location[0]\t\t\t\t#chromosome \n",
    "\t\t\tc=location[5]\t\t\t\t#Watson or Crick (+/-)\n",
    "\t\t\td=location[6]\t\t\t\t#BP Position\n",
    "\t\t\tif b==chromosome and d==position:\t\t#Now if chromosomes are the same AND the position is the same as where we are in the coverage file\n",
    "\t\t\t\tif c=='+':\t\t\t#Watson strand\n",
    "\t\t\t\t\th=1\t\t\t\t#positive value\n",
    "\t\t\t\tif c=='-':\t\t\t#crick strand\t\n",
    "\t\t\t\t\th=-1\t\t\t#negative value  (This is because some features like ACS are asymmetric so we can correct for this)\n",
    "\t\t\t\tfor z in range(2*(int(bp))+1):\n",
    "\t\t\t\t\tk=int(d)-int(bp)*h+z*h\t\t\t\t#d is the feature position, we would like to start 1000 before and go 1000 after and adjust for Watson/Crick\n",
    "\t\t\t\t\tl=Norma[j-int(bp)*h+z*h].split()\t#split the file where the position is\n",
    "\t\t\t\t\tA.append([str(b) + '\\t' + str(-int(bp)+z) + '\\t' + l[2] + '\\t' + \t\t\t\t\t\t#chromosome\n",
    "\t\t\t\t\tA.write(b)\t\t\t\t\t\t#chromosome\n",
    "\t\t\t\t\tA.write('\\t')\n",
    "\t\t\t\t\tA.write(str(-int(bp)+z))\t\t#position relative to site\n",
    "\t\t\t\t\tA.write('\\t')\n",
    "\t\t\t\t\tA.write(l[2])\t\t\t\t\t#counts at this position\n",
    "\t\t\t\t\tA.write('\\t')\n",
    "\t\t\t\t\tA.write(str(k))\t\t\t\t\t#Relative Position \n",
    "\t\t\t\t\tA.write('\\t')\n",
    "\t\t\t\t\tA.write(str(k))\t\t\t\t\t#Relative Position Again \n",
    "\t\t\t\t\tA.write('\\t')\n",
    "\t\t\t\t\tA.write(a[1])\t\t\t\t\t#gene name OR Watson/Crick\n",
    "\t\t\t\t\tA.write('\\t')\n",
    "\t\t\t\t\tA.write(c)\t\t\t\t\t\t#W/+ or C/-\n",
    "\t\t\t\t\tA.write('\\n')\n",
    "\t\n",
    "elapsed = time.time()-start\n",
    "print ('part I took ', elapsed, 'seconds')\n",
    "A.close()\n",
    "\n",
    "file2 = open(motif+'_'+mnasedata+'_aligned').readlines()\n",
    "cluster = open(motif+'_'+mnasedata+'_aligned_'+'clustered'+bp+'.sgr', 'w')\t\n",
    "\n",
    "start=time.time()\n",
    "\n",
    "cluster.write('Header\\tNAME\\tGWEIGHT\\t')\n",
    "for i in range(2*(int(bp))+1):\n",
    "\tif math.fabs(-int(bp)+i)<=float(int(bp)):\n",
    "\t\tcluster.write(str(-int(bp)+i))\n",
    "\t\tcluster.write('\\t')\n",
    "cluster.write('\\n')\t\n",
    "cluster.write('EWEIGHT\\t\\t\\t')\n",
    "for i in range(2*(int(bp))+1):\n",
    "\tif math.fabs(-int(bp)+i)<=float(int(bp)):\n",
    "\t\tcluster.write('1')\n",
    "\t\tcluster.write('\\t')\n",
    "cluster.write('\\n')\n",
    "\n",
    "for i in range(len(file2)/(2*(int(bp))+1)):\n",
    "\ta=locations\t\t\t#file is list of genes from which TSS is derived\n",
    "\tcluster.write(a[0])\t\t\t#writes identifying info for TSS site (gene name)\n",
    "\tcluster.write(' ')\n",
    "\tcluster.write(a[1])\n",
    "\tcluster.write(' ')\n",
    "\tcluster.write(a[6])\n",
    "\tcluster.write('\\t')\n",
    "\tcluster.write(a[0])\t\t\t#writes identifying info for TSS site (gene name)\n",
    "\tcluster.write(' ')\n",
    "\tcluster.write(a[1])\n",
    "\tcluster.write(' ')\n",
    "\tcluster.write(a[6])\n",
    "\tcluster.write('\\t')\n",
    "\tcluster.write('1\\t')\n",
    "\n",
    "\tfor j in range(2*(int(bp))+1):\n",
    "\t\tb=file2[i*(2*(int(bp))+1)+j].split()\t#WT \n",
    "\n",
    "\t\tif math.fabs(int(b[1]))<=float(bp):\t#listing to surrounding TSS\n",
    "\t\t\tcluster.write(str(float(b[2])))\t\t\t#writes WT\n",
    "\t\t\tcluster.write('\\t')\n",
    "\t\t\t\n",
    "\tcluster.write('\\n')\n",
    "\n",
    "cluster.close()\t\n",
    "\n",
    "\t\t\t\t\t\n",
    "\n",
    "\n",
    "elapsed2=time.time()-start\n",
    "print ('part II took', elapsed2, ' seconds')\n",
    "\n",
    "print ('the run took ', elapsed+elapsed2, ' seconds')\n",
    "\n",
    "###\n",
    "# OUTPUT: [after specifying +/- 500bp from motif]\n",
    "# [Header] [NAME] [GWEIGHT] [-500] [-499] [...] [499] [500]\n",
    "# [EWEIGHT] []    []        [1]    [1]     [1]   [1]   [1]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plotdata = pd.read_csv(motif+'_'+mnasedata+'_aligned_'+'clustered'+bp+'.sgr', sep='/t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Jeff's script: make the excel sheet ()\n",
    "\n",
    "#This file sorts the HSF1_targets by start site for easier manipulation in subsequent rounds\n",
    "\n",
    "import math\n",
    "import sys\n",
    "import os\n",
    "import time\n",
    "from decimal import *\n",
    "\n",
    "List=open('/projects/mcknightlab/shared/Scripts_From_Hutch/Breeden/'+sys.argv[1]+'/'+sys.argv[1]+'_HITS_SITES.txt').readlines()\t\t#Select region of interest from features list and use prefix here\n",
    "file=open(sys.argv[2]).readlines()\t\t#Input file 2 is a genomic coverage file of nucleosome positions\n",
    "A=open(sys.argv[3]+'_'+sys.argv[1]+'_aligned', 'w')\t\t\t\t#Output file is the location of aligned positions\n",
    "\n",
    "start=time.time()\n",
    "\n",
    "Chr=[]\n",
    "\t\n",
    "for i in range(len(List)):\t\t#First searches through list of positions in question ie ACS locations\n",
    "\ta=List[i].split()\t\t\t\n",
    "\tChr.append(a[6])\t\t\t#Dumps the location (irrespective of chromosome) into a list\n",
    "\t\n",
    "\n",
    "\t\n",
    "\t\n",
    "for j in range(len(file)):\t#Searches through entire normalized MNase coverage file\n",
    "\te=file[j].split()\n",
    "\tf=e[0]\t\t\t\t\t#chromosome\n",
    "\tg=e[1]\t\t\t\t\t#position\n",
    "\tif g in Chr:\t\t\t#if the position is one of the positions in the list (irrespective of chromosome)\n",
    "\t\tfor i in range(len(List)):\t#program searches through the list of features again (ie ACS locations)\n",
    "\t\t\ta=List[i].split()\n",
    "\t\t\tb=a[0]\t\t\t\t#chromosome \n",
    "\t\t\tc=a[5]\t\t\t\t#Watson or Crick (+/-)\n",
    "\t\t\td=a[6]\t\t\t\t#Position within the list\n",
    "\t\t\tif b==f and d==g:\t\t#Now if chromosomes are the same AND the position is the same as where we are in the coverage file\n",
    "\t\t\t\tif c=='+':\t\t\t#Watson strand\n",
    "\t\t\t\t\th=1\t\t\t\t#positive value\n",
    "\t\t\t\tif c=='-':\t\t\t#crick strand\t\n",
    "\t\t\t\t\th=-1\t\t\t#negative value  (This is because some features like ACS are asymmetric so we can correct for this)\n",
    "\t\t\t\tfor z in range(2*(int(sys.argv[4]))+1):\n",
    "\t\t\t\t\tk=int(d)-int(sys.argv[4])*h+z*h\t\t\t\t#d is the feature position, we would like to start 1000 before and go 1000 after and adjust for Watson/Crick\n",
    "\t\t\t\t\tl=file[j-int(sys.argv[4])*h+z*h].split()\t#split the file where the position is\n",
    "\t\t\t\t\tA.write(b)\t\t\t\t\t\t#chromosome\n",
    "\t\t\t\t\tA.write('\\t')\n",
    "\t\t\t\t\tA.write(str(-int(sys.argv[4])+z))\t\t#position relative to site\n",
    "\t\t\t\t\tA.write('\\t')\n",
    "\t\t\t\t\tA.write(l[2])\t\t\t\t\t#counts at this position\n",
    "\t\t\t\t\tA.write('\\t')\n",
    "\t\t\t\t\tA.write(str(k))\t\t\t\t\t#Relative Position \n",
    "\t\t\t\t\tA.write('\\t')\n",
    "\t\t\t\t\tA.write(str(k))\t\t\t\t\t#Relative Position Again \n",
    "\t\t\t\t\tA.write('\\t')\n",
    "\t\t\t\t\tA.write(a[1])\t\t\t\t\t#gene name OR Watson/Crick\n",
    "\t\t\t\t\tA.write('\\t')\n",
    "\t\t\t\t\tA.write(c)\t\t\t\t\t\t#W/+ or C/-\n",
    "\t\t\t\t\tA.write('\\n')\n",
    "\t\n",
    "elapsed = time.time()-start\n",
    "print 'part I took ', elapsed, 'seconds'\n",
    "A.close()\n",
    "\n",
    "file2 = open(sys.argv[3]+'_'+sys.argv[1]+'_aligned').readlines()\n",
    "cluster = open(sys.argv[3]+'_'+sys.argv[1]+'_aligned_'+'clustered'+sys.argv[4]+'.sgr', 'w')\t\n",
    "\n",
    "start=time.time()\n",
    "\n",
    "cluster.write('Header\\tNAME\\tGWEIGHT\\t')\n",
    "for i in range(2*(int(sys.argv[4]))+1):\n",
    "\tif math.fabs(-int(sys.argv[4])+i)<=float(int(sys.argv[4])):\n",
    "\t\tcluster.write(str(-int(sys.argv[4])+i))\n",
    "\t\tcluster.write('\\t')\n",
    "cluster.write('\\n')\t\n",
    "cluster.write('EWEIGHT\\t\\t\\t')\n",
    "for i in range(2*(int(sys.argv[4]))+1):\n",
    "\tif math.fabs(-int(sys.argv[4])+i)<=float(int(sys.argv[4])):\n",
    "\t\tcluster.write('1')\n",
    "\t\tcluster.write('\\t')\n",
    "cluster.write('\\n')\n",
    "\n",
    "for i in range(len(file2)/(2*(int(sys.argv[4]))+1)):\n",
    "\ta=List[i].split()\t\t\t#file is list of genes from which TSS is derived\n",
    "\tcluster.write(a[0])\t\t\t#writes identifying info for TSS site (gene name)\n",
    "\tcluster.write(' ')\n",
    "\tcluster.write(a[1])\n",
    "\tcluster.write(' ')\n",
    "\tcluster.write(a[6])\n",
    "\tcluster.write('\\t')\n",
    "\tcluster.write(a[0])\t\t\t#writes identifying info for TSS site (gene name)\n",
    "\tcluster.write(' ')\n",
    "\tcluster.write(a[1])\n",
    "\tcluster.write(' ')\n",
    "\tcluster.write(a[6])\n",
    "\tcluster.write('\\t')\n",
    "\tcluster.write('1\\t')\n",
    "\n",
    "\tfor j in range(2*(int(sys.argv[4]))+1):\n",
    "\t\tb=file2[i*(2*(int(sys.argv[4]))+1)+j].split()\t#WT \n",
    "\n",
    "\t\tif math.fabs(int(b[1]))<=float(sys.argv[4]):\t#listing to surrounding TSS\n",
    "\t\t\tcluster.write(str(float(b[2])))\t\t\t#writes WT\n",
    "\t\t\tcluster.write('\\t')\n",
    "\t\t\t\n",
    "\tcluster.write('\\n')\n",
    "\n",
    "cluster.close()\t\n",
    "\n",
    "\t\t\t\t\t\n",
    "\n",
    "\n",
    "elapsed2=time.time()-start\n",
    "print 'part II took', elapsed2, ' seconds'\n",
    "\n",
    "print 'the run took ', elapsed+elapsed2, ' seconds'\n",
    "\n",
    "\n",
    "###\n",
    "# OUTPUT: [after specifying +/- 500bp from motif]\n",
    "# [Header] [NAME] [GWEIGHT] [-500] [-499] [...] [499] [500]\n",
    "# [EWEIGHT] []    []        [1]    [1]     [1]   [1]   [1]"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
